spec/forms/ctc/spouse_filed2019_form_spec.rb

```diff
@@ -0,0 +1,87 @@
+require "rails_helper"
+
+describe Ctc::SpouseFiled2019Form do
+  let(:client) { create :client, tax_returns: [(create :tax_return, filing_status: nil )]}
+  let!(:intake) { create :ctc_intake, client: client }
+  let(:params) { { spouse_filed_2019: filed_2019 } }
+  let(:filed_2019) { "filed_full_joint" }
+
+  context "validations" do
+    context "when filing status is selected" do
+      it "is valid" do
+        expect(
+            described_class.new(intake, params)
+        ).to be_valid
+      end
+    end
+
+    context "when spouse_filed_2019 is not filled in" do
+      it "is not valid" do
+        expect(
+            described_class.new(intake, {})
+        ).not_to be_valid
+      end
+    end
+
+    context "when spouse_filed_2019 is not in the set" do
+      let(:filed_2019) { "on_the_moon" }
+      it "is not valid" do
+        expect {
+          described_class.new(intake, params).save
+        }.to raise_error ArgumentError
+      end
+    end
+  end
+
+  context "save" do
+    context "filed_full_joint" do
+      let(:filed_2019) { "filed_full_joint" }
+      it "updates spouse filed 2019, does not set agi" do
```

Comment:
These tests have a lot of boilerplate, which makes it hard to know what's unique about each test case. Consider a `before ... do` with `described_class.new(intake, params).save`.

I know that doing that makes it hard to use the `expect { ... }.to change(...)` syntax. I would rather see:

```
      let(:filed_2019) { "filed_full_joint" }
      it "updates spouse filed 2019, does not set agi" do
         intake.reload
         expect(intake.spouse_filed_2019).to eq("filed_full_joint")
         expect(intake.spouse_prior_year_agi_amount).to eq(nil)
      end
```

The tests are all so similar that I wish it weren't necessary to write the "it ..." description name at all. Maybe this is a good use for some of these ideas:

https://stackoverflow.com/questions/8754582/best-way-to-organize-tests-in-rspec-that-have-a-combinations-of-factor

Anyway, for now, please do fix the test description names, or come up with a better way to organize these! :)